components/brave_wallet/browser/eth_json_rpc_controller.cc

```diff
@@ -249,6 +252,44 @@ void EthJsonRpcController::OnGetERC20TokenBalance(
   std::move(callback).Run(true, result);
 }
 
+bool EthJsonRpcController::EnsProxyReaderResolveAddress(
+    const std::string& contract_address,
+    const std::string& domain,
+    const std::vector<std::string>& keys,
+    UnstoppableDomainsProxyReaderGetManyCallback callback) {
+  auto internal_callback =
+      base::BindOnce(&EthJsonRpcController::OnEnsProxyReaderResolveAddress,
+                     base::Unretained(this), std::move(callback));
+  std::string data;
+  if (!ens::GetContentHashAddress(domain, &data)) {
+    return false;
+  }
+
+  Request(eth_call("", contract_address, "", "", "", data, "latest"),
+          std::move(internal_callback), true);
+  return true;
+}
+
+void EthJsonRpcController::OnEnsProxyReaderResolveAddress(
+    UnstoppableDomainsProxyReaderGetManyCallback callback,
+    const int status,
+    const std::string& body,
+    const std::map<std::string, std::string>& headers) {
+  if (status < 200 || status > 299) {
+    if (callback)
```

Comment:
why check callback? it seems it is always valid. Maybe just DCHECK in the very beginning of the method

Reply:
this way a single `if` should be enough? 
I'd rather modify tests to make them pass a stub, than add a lot of boilerplate to production code

Reply:
np, done